snuba/util.py

```diff
@@ -444,7 +444,11 @@ def raw_query(body, sql, client, timer, stats=None):
                             query_settings['max_threads'] = 1
 
                         try:
-                            result = NativeDriverReader(client).execute(
+                            result = HTTPReader(
```

Comment:
We'll probably need a way to find a way to switch these out at runtime, since this is a bit more touchy than the writer change. The only way that I've thought of testing this (short of Matt doing some wacky Kubernetes stuff) is to (random?) sample in requests based on dynamic configurations. Looking for additional ideas here.

Reply:
Doing something like serviceDelegator and run both for a small number of queries? That will provide you a more solid validation in a shorter time. The fact reader is a class on its own helps:

```
class DelegatingReader:
     def __init__(self, native, http):
         ...
 
     def execute ():
          if random < settings.EXPERIMENT_SAMPLE:
              native.execute
              http.execute 
              compare and log discrepancy 
              return native result if discrepancy is too big
         else:
              if settings.PROD_READER = 'native:
                  return native.execute
              else:
                  return http.execute
```

You can add a fallback to native in case of error on http so we do not take down Snuba.